libraries/AP_Scripting/lua_scripts.cpp

```diff
@@ -157,7 +149,7 @@ void lua_scripts::load_all_scripts_in_dir(lua_State *L, const char *dirname) {
         reschedule_script(script);
 
     }
-    closedir(d);
+    AP::FS().closedir(d);
```

Comment:
We may never hit this, because load_script may never return here if there is an error within the script. This would leak the dir. I think we can resolve this by making the pointer be a part of the `lua_scripts` class, then in the panic handler we can check if it's not `nullptr` and delete it then. (As long as we always set it to be `nullptr` after deleting it). This is a small leak at the moment, but in the future we are considering reloading scripts more frequently.

Reply:
or we could add a AP::FS() method that does does a opendir with a pre-allocated piece of memory or with a realloc() function as a univeral allocator fn

Reply:
@WickedShell isn't this an existing bug anyway? (at least on HAL_Linux, where opendir does alloc)